app/Http/Controllers/AudioController.php

```diff
@@ -25,7 +25,31 @@ public function create(Request $request): View
 
     public function store(AudioStoreRequest $request): RedirectResponse
     {
-        $audio = Audio::create($request->validated());
+        $validated = $request->validated();
+        $audio = [
+            'name' => $validated['name'],
+            'author' => $validated['artist'],
+            'user_id' => auth()->id(),
+        ];
+
+        $audio['disk'] = config('filesystems.default');
+        $audio['path'] = $request->file('file')->store('audios', $audio['disk']);
+
+        $getID3 = new \getID3;
+        $file = $getID3->analyze(storage_path('app') . '/' .$audio['path']);
```

Comment:
I was looking in [web](https://stackoverflow.com/questions/46854142/laravel-getid3-unable-to-pull-file-from-s3) and maybe we need refactor this if we use external disk